ArduCopter/GCS_Copter.cpp

```diff
@@ -116,22 +116,24 @@ void GCS_Copter::update_vehicle_sensor_status_flags(void)
 
 #if OPTFLOW == ENABLED
     const OpticalFlow *optflow = AP::opticalflow();
-    if (optflow && optflow->enabled()) {
-        control_sensors_present |= MAV_SYS_STATUS_SENSOR_OPTICAL_FLOW;
-        control_sensors_enabled |= MAV_SYS_STATUS_SENSOR_OPTICAL_FLOW;
-    }
-    if (optflow && optflow->healthy()) {
-        control_sensors_health |= MAV_SYS_STATUS_SENSOR_OPTICAL_FLOW;
```

Comment:
This exact same code is present in Plane *and* Sub.  After this change we'll have two different versions of the code.

I do prefer your new formulation here.

.... but I also think we should probably put OpticalFlow on AP_Vehicle, especially now we can play with scheduler tables priority a bit better.